In [209]:
import pandas as pd
pd.set_option('display.max_columns', None)

In [210]:
indices_df = pd.read_csv("../data/raw/indices data.csv", index_col=False)
surface_df = pd.read_csv("../data/raw/surface_data.csv", index_col=False)

In [211]:
indices_df.head()

,Year,Month,Date,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob
0,1981,1,1,0,91.2,12.4,13.4,-1.4,3.0,21.7,24.7,284.41,993.98,0.0,0.0,22.8,5636,NaN
1,1981,1,1,12,66.0,9.3,8.0,0.0,6.2,22.7,28.9,281.75,856.09,0.0,0.0,24.7,5671,NaN
2,1981,1,2,0,75.7,10.2,11.3,1.6,10.2,20.1,30.3,282.11,956.13,0.0,0.0,20.4,5592,NaN
3,1981,1,2,12,111.7,8.5,7.5,0.8,13.1,20.1,33.2,280.44,836.26,0.0,0.0,23.0,5660,NaN
4,1981,1,3,12,64.0,2.1,-0.2,2.8,13.4,24.1,37.5,284.25,862.29,-260.5,0.8,19.7,5636,NaN


In [212]:
indices_df.columns

Index(['Year', 'Month', 'Date', 'GMT', 'SWEAT index', 'Showalter index',
       'LIFTED index', 'K index', 'Cross totals index',
       'Vertical totals index', 'Totals totals index', 'TLCL', 'PLCL', 'CINE',
       'CAPE', 'PRECIPITABLE WATER', '1000-500 THICKNESS', 'prob'],
      dtype='object')

In [213]:
surface_df.head()

,INDEX,YEAR,MN,DT,MAX,MIN,AW,RF,EVP,DRNRF(hrs),DRNRF(mnts),SSH,TS,N,FFF,D,TOCSQ,DU,RA,DZ,SN,SL,HA,TH,DS,FG,GA,T,G,DUR,T.1,G.1,DUR.1,T.2,G.2,DUR.2,T.3,G.3,DUR.3,Unnamed: 39
0,42809,1981,1,1,26.0,12.1,04,0.0,,,,,,,,,,,0,0,0,0,0,0,0,0,0,2,2,180,,,,,,,,,,
1,42809,1981,1,2,25.0,11.6,03,0.0,,,,,,,,,,,0,0,0,0,0,0,0,1,0,4,2,195,2,4,045,,,,,,,
2,42809,1981,1,3,25.4,11.3,03,0.0,,,,,,,,,,,0,0,0,0,0,0,0,1,0,4,2,180,2,2,260,,,,,,,
3,42809,1981,1,4,25.8,11.8,02,0.0,,,,,,,,,,,0,0,0,0,0,0,0,0,0,4,3,120,2,2,150,,,,,,,
4,42809,1981,1,5,28.2,13.9,04,0.0,,,,,,,,,,,0,0,0,0,0,0,0,0,0,9,8,030,,,,,,,,,,


- We need to extract morning data from indices_df, specifically where GMT == 0 and focus only on TH at the surface level to determine whether a thunderstorm has occurred.

In [214]:
# copy of df
indices_df_copy = indices_df.copy()
surface_df_copy = surface_df.copy()

In [215]:
indices_df_copy.columns

Index(['Year', 'Month', 'Date', 'GMT', 'SWEAT index', 'Showalter index',
       'LIFTED index', 'K index', 'Cross totals index',
       'Vertical totals index', 'Totals totals index', 'TLCL', 'PLCL', 'CINE',
       'CAPE', 'PRECIPITABLE WATER', '1000-500 THICKNESS', 'prob'],
      dtype='object')

In [216]:
indices_df_copy = indices_df_copy[indices_df_copy["GMT"] == 0]
indices_df_copy.head()

,Year,Month,Date,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob
0,1981,1,1,0,91.2,12.4,13.4,-1.4,3.0,21.7,24.7,284.41,993.98,0.0,0.0,22.8,5636,NaN
2,1981,1,2,0,75.7,10.2,11.3,1.6,10.2,20.1,30.3,282.11,956.13,0.0,0.0,20.4,5592,NaN
5,1981,1,4,0,128.3,6.9,6.9,17.5,16.3,25.3,41.6,284.79,978.71,0.0,0.0,23.8,5581,NaN
7,1981,1,5,0,194.2,2.8,1.2,23.5,22.5,28.1,50.6,286.68,965.77,-138.5,83.1,28.6,5578,NaN
9,1981,1,6,0,151.4,5.1,1.8,25.4,18.9,24.9,43.8,289.11,975.72,-112.9,47.9,33.2,5648,NaN


In [217]:
len(indices_df_copy)

11883

In [218]:
indices_df_copy.isnull().sum()

Year                         0
Month                        0
Date                         0
GMT                          0
SWEAT index                  0
Showalter index              0
LIFTED index                 0
K index                      0
Cross totals index           0
Vertical totals index        0
Totals totals index          0
TLCL                         0
PLCL                         0
CINE                         0
CAPE                         0
PRECIPITABLE WATER           0
1000-500 THICKNESS           0
prob                     11577
dtype: int64

In [219]:
surface_df_copy.isnull().sum().head(6)

INDEX     0
YEAR      0
MN        0
DT        0
MAX      37
MIN      20
dtype: int64

In [220]:
indices_df_copy.rename(columns={"Date": "Day"}, inplace=True)

In [221]:
indices_df_copy.head()

,Year,Month,Day,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob
0,1981,1,1,0,91.2,12.4,13.4,-1.4,3.0,21.7,24.7,284.41,993.98,0.0,0.0,22.8,5636,NaN
2,1981,1,2,0,75.7,10.2,11.3,1.6,10.2,20.1,30.3,282.11,956.13,0.0,0.0,20.4,5592,NaN
5,1981,1,4,0,128.3,6.9,6.9,17.5,16.3,25.3,41.6,284.79,978.71,0.0,0.0,23.8,5581,NaN
7,1981,1,5,0,194.2,2.8,1.2,23.5,22.5,28.1,50.6,286.68,965.77,-138.5,83.1,28.6,5578,NaN
9,1981,1,6,0,151.4,5.1,1.8,25.4,18.9,24.9,43.8,289.11,975.72,-112.9,47.9,33.2,5648,NaN


In [222]:
surface_df_copy.rename(columns={"YEAR": "Year","MN": "Month", "DT": "Day"}, inplace=True)

In [223]:
indices_df_copy["Date"] = pd.to_datetime(indices_df_copy[["Year", "Month", "Day"]]).dt.strftime('%d/%m/%Y')
surface_df_copy["Date"] = pd.to_datetime(surface_df_copy[["Year", "Month", "Day"]]).dt.strftime('%d/%m/%Y')

In [224]:
indices_df_copy.head()

,Year,Month,Day,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob,Date
0,1981,1,1,0,91.2,12.4,13.4,-1.4,3.0,21.7,24.7,284.41,993.98,0.0,0.0,22.8,5636,NaN,01/01/1981
2,1981,1,2,0,75.7,10.2,11.3,1.6,10.2,20.1,30.3,282.11,956.13,0.0,0.0,20.4,5592,NaN,02/01/1981
5,1981,1,4,0,128.3,6.9,6.9,17.5,16.3,25.3,41.6,284.79,978.71,0.0,0.0,23.8,5581,NaN,04/01/1981
7,1981,1,5,0,194.2,2.8,1.2,23.5,22.5,28.1,50.6,286.68,965.77,-138.5,83.1,28.6,5578,NaN,05/01/1981
9,1981,1,6,0,151.4,5.1,1.8,25.4,18.9,24.9,43.8,289.11,975.72,-112.9,47.9,33.2,5648,NaN,06/01/1981


In [225]:
(- len(indices_df_copy) + len(surface_df_copy)) / len(surface_df_copy) *100

17.46197124400917

- around 17% of data is missing so GMT == 0 is not a good approach

In [226]:
# copy of df
indices_df_copy = indices_df.copy()
surface_df_copy = surface_df.copy()

In [227]:
indices_df_copy.isnull().sum()

Year                         0
Month                        0
Date                         0
GMT                          0
SWEAT index                  0
Showalter index              0
LIFTED index                 0
K index                      0
Cross totals index           0
Vertical totals index        0
Totals totals index          0
TLCL                         0
PLCL                         0
CINE                         0
CAPE                         0
PRECIPITABLE WATER           0
1000-500 THICKNESS           0
prob                     22018
dtype: int64

In [228]:
indices_df_copy.head()

,Year,Month,Date,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob
0,1981,1,1,0,91.2,12.4,13.4,-1.4,3.0,21.7,24.7,284.41,993.98,0.0,0.0,22.8,5636,NaN
1,1981,1,1,12,66.0,9.3,8.0,0.0,6.2,22.7,28.9,281.75,856.09,0.0,0.0,24.7,5671,NaN
2,1981,1,2,0,75.7,10.2,11.3,1.6,10.2,20.1,30.3,282.11,956.13,0.0,0.0,20.4,5592,NaN
3,1981,1,2,12,111.7,8.5,7.5,0.8,13.1,20.1,33.2,280.44,836.26,0.0,0.0,23.0,5660,NaN
4,1981,1,3,12,64.0,2.1,-0.2,2.8,13.4,24.1,37.5,284.25,862.29,-260.5,0.8,19.7,5636,NaN


In [229]:
indices_df_copy.rename(columns={"Date": "Day"}, inplace=True)

In [230]:
indices_df_copy["Date"] = pd.to_datetime(indices_df_copy[["Year", "Month", "Day"]]).dt.strftime('%d/%m/%Y')

In [231]:
indices_df_copy.head(7)

,Year,Month,Day,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob,Date
0,1981,1,1,0,91.2,12.4,13.4,-1.4,3.0,21.7,24.7,284.41,993.98,0.0,0.0,22.8,5636,NaN,01/01/1981
1,1981,1,1,12,66.0,9.3,8.0,0.0,6.2,22.7,28.9,281.75,856.09,0.0,0.0,24.7,5671,NaN,01/01/1981
2,1981,1,2,0,75.7,10.2,11.3,1.6,10.2,20.1,30.3,282.11,956.13,0.0,0.0,20.4,5592,NaN,02/01/1981
3,1981,1,2,12,111.7,8.5,7.5,0.8,13.1,20.1,33.2,280.44,836.26,0.0,0.0,23.0,5660,NaN,02/01/1981
4,1981,1,3,12,64.0,2.1,-0.2,2.8,13.4,24.1,37.5,284.25,862.29,-260.5,0.8,19.7,5636,NaN,03/01/1981
5,1981,1,4,0,128.3,6.9,6.9,17.5,16.3,25.3,41.6,284.79,978.71,0.0,0.0,23.8,5581,NaN,04/01/1981
6,1981,1,4,12,98.8,1.9,0.3,11.2,15.5,28.3,43.8,281.81,830.76,0.0,0.0,19.3,5625,NaN,04/01/1981


In [232]:
indices_df_copy["Date"].duplicated().sum()

np.int64(10374)

In [233]:
indices_df_copy.drop_duplicates(subset=["Date"], inplace=True)

In [234]:
len(indices_df_copy)

12075

In [235]:
indices_df_copy.head()

,Year,Month,Day,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob,Date
0,1981,1,1,0,91.2,12.4,13.4,-1.4,3.0,21.7,24.7,284.41,993.98,0.0,0.0,22.8,5636,NaN,01/01/1981
2,1981,1,2,0,75.7,10.2,11.3,1.6,10.2,20.1,30.3,282.11,956.13,0.0,0.0,20.4,5592,NaN,02/01/1981
4,1981,1,3,12,64.0,2.1,-0.2,2.8,13.4,24.1,37.5,284.25,862.29,-260.5,0.8,19.7,5636,NaN,03/01/1981
5,1981,1,4,0,128.3,6.9,6.9,17.5,16.3,25.3,41.6,284.79,978.71,0.0,0.0,23.8,5581,NaN,04/01/1981
7,1981,1,5,0,194.2,2.8,1.2,23.5,22.5,28.1,50.6,286.68,965.77,-138.5,83.1,28.6,5578,NaN,05/01/1981


In [236]:
indices_df_copy.tail()

,Year,Month,Day,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob,Date
22439,2020,12,27,0,75.7,17.1,17.2,-19.9,-24.3,21.8,-2.5,279.17,876.91,0.0,0.0,9.5,5731,NaN,27/12/2020
22441,2020,12,28,0,67.9,15.9,15.5,-6.5,-21.5,20.4,-1.1,280.77,893.60,0.0,0.0,11.4,5718,NaN,28/12/2020
22443,2020,12,29,0,98.9,16.2,16.1,-15.2,5.3,18.1,23.4,278.86,862.08,0.0,0.0,13.4,5717,NaN,29/12/2020
22445,2020,12,30,0,135.8,13.4,12.2,-6.5,11.4,18.1,29.5,282.07,892.90,0.0,0.0,16.2,5715,NaN,30/12/2020
22447,2020,12,31,0,97.0,23.8,26.2,-4.2,1.2,21.3,22.5,260.81,698.99,0.0,0.0,13.9,5720,NaN,31/12/2020


In [237]:
len(surface_df_copy)

14397

In [238]:
full_range = pd.date_range(start="01/01/1981", end="31/12/2020")
missing_dates = full_range.difference(indices_df_copy["Date"])
len(missing_dates)

14610

In [239]:
print(indices_df_copy["Date"].dtype)

object


In [240]:
indices_df_copy["Date"] = pd.to_datetime(indices_df_copy["Date"], format="%d/%m/%Y", errors="coerce")

In [241]:
print(indices_df_copy["Date"].dtype)

datetime64[ns]


In [242]:
full_range = pd.date_range(start="01/01/1981", end="31/12/2020")
missing_dates = full_range.difference(indices_df_copy["Date"])
len(missing_dates)

2535

In [243]:
missing_dates

DatetimeIndex(['1981-04-19', '1981-08-20', '1981-08-21', '1981-08-22',
               '1981-08-23', '1981-12-26', '1981-12-27', '1981-12-28',
               '1981-12-29', '1981-12-30',
               ...
               '2020-03-05', '2020-03-06', '2020-03-07', '2020-03-08',
               '2020-03-09', '2020-03-10', '2020-04-12', '2020-04-13',
               '2020-05-28', '2020-05-29'],
              dtype='datetime64[ns]', length=2535, freq=None)

In [244]:
indices_df_copy[105:110]

,Year,Month,Day,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob,Date
204,1981,4,16,0,231.9,2.6,-4.9,31.6,23.1,25.5,48.6,296.06,982.65,-23.0,2690.2,43.2,5743,NaN,1981-04-16
206,1981,4,17,0,193.9,4.3,-0.7,35.5,17.5,31.9,49.4,291.49,977.98,-374.1,91.5,37.2,5744,NaN,1981-04-17
208,1981,4,18,0,850.7,-2.8,-8.3,47.3,32.3,35.3,67.6,298.07,985.06,-104.6,19.6,57.1,5730,NaN,1981-04-18
210,1981,4,20,12,391.7,-1.6,-7.9,20.4,27.5,28.5,56.0,293.63,923.87,-68.6,1647.0,38.6,5701,NaN,1981-04-20
211,1981,4,21,0,139.0,5.1,-1.4,7.2,17.1,24.1,41.2,294.33,990.31,-119.9,627.5,34.2,5757,NaN,1981-04-21


In [245]:
len(missing_dates)/len(full_range) *100

17.35112936344969

- I checked both the raw data and the DataFrame, and the date 19/04/1981 is indeed missing. Overall, about 2,535 dates are missing, which is roughly 17% of the indices data.
- Both techniques result in the same percentage of data loss.

In [246]:
indices_df_copy.head()

,Year,Month,Day,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob,Date
0,1981,1,1,0,91.2,12.4,13.4,-1.4,3.0,21.7,24.7,284.41,993.98,0.0,0.0,22.8,5636,NaN,1981-01-01
2,1981,1,2,0,75.7,10.2,11.3,1.6,10.2,20.1,30.3,282.11,956.13,0.0,0.0,20.4,5592,NaN,1981-01-02
4,1981,1,3,12,64.0,2.1,-0.2,2.8,13.4,24.1,37.5,284.25,862.29,-260.5,0.8,19.7,5636,NaN,1981-01-03
5,1981,1,4,0,128.3,6.9,6.9,17.5,16.3,25.3,41.6,284.79,978.71,0.0,0.0,23.8,5581,NaN,1981-01-04
7,1981,1,5,0,194.2,2.8,1.2,23.5,22.5,28.1,50.6,286.68,965.77,-138.5,83.1,28.6,5578,NaN,1981-01-05


In [247]:
surface_df_copy.rename(columns={"YEAR": "Year", "MN": "Month", "DT": "Day"}, inplace=True)

In [248]:
surface_df_copy.head()

,INDEX,Year,Month,Day,MAX,MIN,AW,RF,EVP,DRNRF(hrs),DRNRF(mnts),SSH,TS,N,FFF,D,TOCSQ,DU,RA,DZ,SN,SL,HA,TH,DS,FG,GA,T,G,DUR,T.1,G.1,DUR.1,T.2,G.2,DUR.2,T.3,G.3,DUR.3,Unnamed: 39
0,42809,1981,1,1,26.0,12.1,04,0.0,,,,,,,,,,,0,0,0,0,0,0,0,0,0,2,2,180,,,,,,,,,,
1,42809,1981,1,2,25.0,11.6,03,0.0,,,,,,,,,,,0,0,0,0,0,0,0,1,0,4,2,195,2,4,045,,,,,,,
2,42809,1981,1,3,25.4,11.3,03,0.0,,,,,,,,,,,0,0,0,0,0,0,0,1,0,4,2,180,2,2,260,,,,,,,
3,42809,1981,1,4,25.8,11.8,02,0.0,,,,,,,,,,,0,0,0,0,0,0,0,0,0,4,3,120,2,2,150,,,,,,,
4,42809,1981,1,5,28.2,13.9,04,0.0,,,,,,,,,,,0,0,0,0,0,0,0,0,0,9,8,030,,,,,,,,,,


In [249]:
surface_df_copy["Date"] = pd.to_datetime(surface_df_copy[["Year", "Month", "Day"]], format="%d/%m/%Y", errors="coerce")

In [250]:
surface_df_copy.head()

,INDEX,Year,Month,Day,MAX,MIN,AW,RF,EVP,DRNRF(hrs),DRNRF(mnts),SSH,TS,N,FFF,D,TOCSQ,DU,RA,DZ,SN,SL,HA,TH,DS,FG,GA,T,G,DUR,T.1,G.1,DUR.1,T.2,G.2,DUR.2,T.3,G.3,DUR.3,Unnamed: 39,Date
0,42809,1981,1,1,26.0,12.1,04,0.0,,,,,,,,,,,0,0,0,0,0,0,0,0,0,2,2,180,,,,,,,,,,,1981-01-01
1,42809,1981,1,2,25.0,11.6,03,0.0,,,,,,,,,,,0,0,0,0,0,0,0,1,0,4,2,195,2,4,045,,,,,,,,1981-01-02
2,42809,1981,1,3,25.4,11.3,03,0.0,,,,,,,,,,,0,0,0,0,0,0,0,1,0,4,2,180,2,2,260,,,,,,,,1981-01-03
3,42809,1981,1,4,25.8,11.8,02,0.0,,,,,,,,,,,0,0,0,0,0,0,0,0,0,4,3,120,2,2,150,,,,,,,,1981-01-04
4,42809,1981,1,5,28.2,13.9,04,0.0,,,,,,,,,,,0,0,0,0,0,0,0,0,0,9,8,030,,,,,,,,,,,1981-01-05


In [251]:
print(surface_df_copy["Date"].dtype)

datetime64[ns]


In [252]:
surface_df_copy.columns

Index(['INDEX', 'Year', 'Month', 'Day', 'MAX', 'MIN', 'AW', 'RF', 'EVP',
       'DRNRF(hrs)', 'DRNRF(mnts)', 'SSH', 'TS', 'N', 'FFF', 'D', 'TOCSQ',
       'DU', 'RA', 'DZ', 'SN', 'SL', 'HA', 'TH', 'DS', 'FG', 'GA', 'T', 'G',
       'DUR', 'T.1', 'G.1', 'DUR.1', 'T.2', 'G.2', 'DUR.2', 'T.3', 'G.3',
       'DUR.3', 'Unnamed: 39', 'Date'],
      dtype='object')

In [253]:
surface_df_copy.drop(columns=['INDEX', 'Year', 'Month', 'Day', 'MAX', 'MIN', 'AW', 'RF', 'EVP', 'DRNRF(hrs)', 'DRNRF(mnts)', 'SSH', 'TS', 'N', 'FFF', 'D', 'TOCSQ', 'DU', 'RA', 'DZ', 'SN', 'SL', 'HA', 'DS', 'FG', 'GA', 'T', 'G', 'DUR', 'T.1', 'G.1', 'DUR.1', 'T.2', 'G.2', 'DUR.2', 'T.3', 'G.3', 'DUR.3', 'Unnamed: 39',], inplace=True)

In [254]:
surface_df_copy.tail()

,TH,Date
14392,0,2020-12-27
14393,0,2020-12-28
14394,0,2020-12-29
14395,0,2020-12-30
14396,0,2020-12-31


- Since the goal is to predict one day in advance, the data has been shifted by one row.

In [255]:
surface_df_copy["TH"] = surface_df_copy["TH"].shift(-1)
surface_df_copy.tail()

,TH,Date
14392,0,2020-12-27
14393,0,2020-12-28
14394,0,2020-12-29
14395,0,2020-12-30
14396,None,2020-12-31


In [256]:
print(surface_df_copy["TH"].dtype)

object


In [257]:
surface_df_copy["TH"].isna().sum()

np.int64(1)

In [258]:
surface_df_copy["TH"].unique()

array(['0', '1', ' ', None], dtype=object)

In [259]:
surface_df_copy["TH"] = pd.to_numeric(surface_df_copy["TH"], errors="coerce")

In [260]:
surface_df_copy.loc[surface_df_copy["TH"].isna(), "TH"] = 0
surface_df_copy["TH"] = surface_df_copy["TH"].astype(int)
print(surface_df_copy.tail())
print(surface_df_copy["TH"].dtype)

       TH       Date
14392   0 2020-12-27
14393   0 2020-12-28
14394   0 2020-12-29
14395   0 2020-12-30
14396   0 2020-12-31
int64


In [261]:
#saving data in csv 
indices_df_copy.to_csv("../data/processed data/index.csv", index=False)
surface_df_copy.to_csv("../data/processed data/surface.csv", index=False)

merging both of the data frame based on the date

In [262]:
merged_df = pd.merge(indices_df_copy, surface_df_copy, on='Date', how='inner')
merged_df.to_csv("../data/processed data/merged_data.csv", index=False)

In [264]:
merged_df.tail()

,Year,Month,Day,GMT,SWEAT index,Showalter index,LIFTED index,K index,Cross totals index,Vertical totals index,Totals totals index,TLCL,PLCL,CINE,CAPE,PRECIPITABLE WATER,1000-500 THICKNESS,prob,Date,TH
11868,2020,12,27,0,75.7,17.1,17.2,-19.9,-24.3,21.8,-2.5,279.17,876.91,0.0,0.0,9.5,5731,NaN,2020-12-27,0
11869,2020,12,28,0,67.9,15.9,15.5,-6.5,-21.5,20.4,-1.1,280.77,893.60,0.0,0.0,11.4,5718,NaN,2020-12-28,0
11870,2020,12,29,0,98.9,16.2,16.1,-15.2,5.3,18.1,23.4,278.86,862.08,0.0,0.0,13.4,5717,NaN,2020-12-29,0
11871,2020,12,30,0,135.8,13.4,12.2,-6.5,11.4,18.1,29.5,282.07,892.90,0.0,0.0,16.2,5715,NaN,2020-12-30,0
11872,2020,12,31,0,97.0,23.8,26.2,-4.2,1.2,21.3,22.5,260.81,698.99,0.0,0.0,13.9,5720,NaN,2020-12-31,0
